godot-core/src/task/futures.rs

```diff
@@ -308,8 +334,188 @@ impl<C: WithBaseField, R: ParamTuple + Sync + Send> IntoFuture for &TypedSignal<
+                type Target = Self;
+
+                fn assert_safe_send(self) -> Self::Target {
+                    self
+                }
+            }
+
+            impl IntoMaybeSend for $ty {
+                type Target = Self;
+                fn into_maybe_send(self) -> Self::Target {
+                    self
+                }
+            }
+        )+
+    };
+
+    (tuple; $($ty:ident),*) => {
+        impl<$($ty),*> Sealed for ($($ty,)*) {}
+
+        unsafe impl<$($ty: AssertSafeSend),*> AssertSafeSend for ($($ty,)*) {
+            type Target = ($($ty::Target,)*);
+
+            fn assert_safe_send(self) -> Self::Target {
+                #[allow(non_snake_case)]
+                let ($($ty,)*) = self;
+
+                #[allow(clippy::unused_unit)]
+                ($($ty.assert_safe_send(),)*)
+            }
+        }
+
+        impl<$($ty: IntoMaybeSend),*> IntoMaybeSend for ($($ty,)*) {
+            type Target = ($($ty::Target,)*);
+
+            fn into_maybe_send(self) -> Self::Target {
+                #[allow(non_snake_case)]
+                let ($($ty,)*) = self;
+
+                #[allow(clippy::unused_unit)]
+                ($($ty.into_maybe_send(),)*)
+            }
+        }
+    };
+
+    (!Send; $($ty:ty),+) => {
+        $(
+            impl Sealed for $ty {}
+            impl Sealed for ThreadCrosser<$ty> {}
+
+            unsafe impl AssertSafeSend for ThreadCrosser<$ty> {
+                type Target = $ty;
+
+                fn assert_safe_send(self) -> Self::Target {
+                    self.extract()
+                }
+            }
+
+            impl IntoMaybeSend for $ty {
+                type Target = ThreadCrosser<$ty>;
+
+                fn into_maybe_send(self) -> Self::Target {
+                    ThreadCrosser::new(self)
+                }
+            }
+        )+
+    };
+}
+
+impl_maybe_send!(
+    Send;
+    bool, u8, u16, u32, u64, i8, i16, i32, i64, f32, f64, StringName, Transform2D, Transform3D, Vector2, Vector2i, Vector2Axis,
+    Vector3, Vector3i, Vector3Axis, Vector4, Vector4i, Rect2, Rect2i, Plane, Quaternion, Aabb, Basis, Projection, Color, Rid
+);
+
+impl_maybe_send!(
+    !Send;
+    Variant, GString, Dictionary, VariantArray, Callable, NodePath, PackedByteArray, PackedInt32Array, PackedInt64Array, PackedFloat32Array,
+    PackedFloat64Array, PackedStringArray, PackedVector2Array, PackedVector3Array, PackedColorArray, PackedVector4Array, Signal
+);
+
+// This should be kept in sync with crate::registry::signal::variadic.
+impl_maybe_send!(tuple; );
+impl_maybe_send!(tuple; A1);
+impl_maybe_send!(tuple; A1, A2);
+impl_maybe_send!(tuple; A1, A2, A3);
+impl_maybe_send!(tuple; A1, A2, A3, A4);
+impl_maybe_send!(tuple; A1, A2, A3, A4, A5);
+impl_maybe_send!(tuple; A1, A2, A3, A4, A5, A6);
+impl_maybe_send!(tuple; A1, A2, A3, A4, A5, A6, A7);
+impl_maybe_send!(tuple; A1, A2, A3, A4, A5, A6, A7, A8);
+impl_maybe_send!(tuple; A1, A2, A3, A4, A5, A6, A7, A8, A9);
```

Comment:
> This should be kept in sync with crate::registry::signal::variadic.

Then maybe we could add it directly to those macros.

Of course it would mean that the responsibility of `Send` is now mixed with other responsibilities, but this coupling exists either way -- it's just that here it's implicit and only with a comment.

And it would play into my previous comment regarding centralization of static type properties.

Reply:
You resolved this but it's still there -- I assume, we would try to centralize it later, maybe with #1042...

Reply:
Ah I see, to me, this fell under:

> Try to keep changes there as minimal. We can then at a later point refactor the other lists...

I.e. we will merge `crate::registry::signal::variadic` into `crate::bulitin::variant::impls` in a later PR.

Reply:
Sounds good! Maybe wait for #1042, otherwise there will likely be merge conflicts.